storage/database/src/env.rs

```diff
@@ -246,32 +259,22 @@ where
     /// This will open the database [`Table`]
     /// passed as a generic to this function.
     ///
-    #[doc = doc_table_error!()]
-    fn open_db_rw<T: Table>(&self, tx_rw: &Rw) -> Result<impl DatabaseRw<T>, RuntimeError>;
-
-    /// Open all tables in read/iter mode.
-    ///
-    /// This calls [`EnvInner::open_db_ro`] on all database tables
-    /// and returns a structure that allows access to all tables.
+    /// # Errors
+    /// This will only return [`RuntimeError::Io`] on errors.
     ///
-    #[doc = doc_table_error!()]
-    fn open_tables(&self, tx_ro: &Ro) -> Result<impl TablesIter, RuntimeError> {
-        call_fn_on_all_tables_or_early_return! {
-            Self::open_db_ro(self, tx_ro)
-        }
-    }
+    /// Implementation details: Both `heed` & `redb` backends create
+    /// the table with this function if it does not already exist. For safety and
+    /// clear intent, you should still consider using [`EnvInner::create_db`] instead.
+    fn open_db_rw<T: Table>(&self, tx_rw: &Rw) -> Result<impl DatabaseRw<T>, RuntimeError>;
 
-    /// Open all tables in read-write mode.
+    /// Create a database table.
     ///
-    /// This calls [`EnvInner::open_db_rw`] on all database tables
-    /// and returns a structure that allows access to all tables.
+    /// This will create the database [`Table`]
+    /// passed as a generic to this function.
     ///
-    #[doc = doc_table_error!()]
-    fn open_tables_mut(&self, tx_rw: &Rw) -> Result<impl TablesMut, RuntimeError> {
-        call_fn_on_all_tables_or_early_return! {
-            Self::open_db_rw(self, tx_rw)
-        }
-    }
+    /// # Errors
+    /// This will only return [`RuntimeError::Io`] on errors.
+    fn create_db<T: Table>(&self, tx_rw: &Rw) -> Result<(), RuntimeError>;
```

Comment:
## `cuprate_database::Env` 2/3

There is a new function: `create_db`, it just creates the passed `Table`.

The existing:
- `EnvInner::open_db_ro` will error if the table doesn't exist
- `EnvInner::open_db_rw` will create the table if it doesn't exist. This is nice side effect since both `redb/heed` do this but it is documented for clarity that `create_db` should be used just in case